src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs

```diff
@@ -339,6 +339,12 @@ private void Stop()
                         throw new InvalidOperationException("Site not stopped yet");
                     }
 
+                    if (appPool.WorkerProcesses.Any(wp => wp.State == WorkerProcessState.Running ||
```

Comment:
This is because we sometimes force a recycle so HostProcess is not always the running worker process.